src/sparsevector.jl

```diff
@@ -1308,6 +1308,38 @@ hvcat(rows::Tuple{Vararg{Int}}, n1::Number, ns::Vararg{Number}) = invoke(hvcat,
 hvcat(rows::Tuple{Vararg{Int}}, n1::N, ns::Vararg{N}) where {N<:Number} = invoke(hvcat, Tuple{typeof(rows), Vararg{N}}, rows, n1, ns...)
 
 
+### Efficient repetition of sparse vectors
+
+function Base.repeat(v::AbstractSparseVector, m)
+    nnz_source = nnz(v)
+    nnz_new = nnz_source * m
+
+    nzind_source, nzval_source = findnz(v)
```

Comment:
avoid using `findnz`, it's allocating. How about
```julia
    nzind = similar(nonzeroinds(v), nnz_new)
    nzval = similar(nonzeros(v), nnz_new)

    ptr_res = 1
    for index_repetition = 0:(m-1)
        row_offset = index_repetition * length(v)
        stuffcol!(nzind, nzval, ptr_res, nonzeroinds(v), nonzeros(v), 1, nnz_source, row_offset)
        ptr_res += nnz_source
    end
```
it uses 4 fewer allocations

Reply:
Thanks a lot, I did not realize `findnz` does a copy, I thought it's just a convenient way to get essentially both `nonzeroinds` and `nonzeros`. Fixed in b961a7f

Reply:
perfect